docs/source/Technical-Glossary.rst

```diff
@@ -0,0 +1,212 @@
+PDFs, locating the recipient, etc., the message producer only has to upload their message and instructions to the queue
+and RabbitMQ will take care of the rest. Using this service makes messaging through Lookit easier and more efficient, as
+it is able to re-queue messages, it is faster and more reliable, and it is scalable for when there are a lot of messages.
+
+`source <https://www.cloudamqp.com/blog/2015-05-18-part1-rabbitmq-for-beginners-what-is-rabbitmq.html>`_
+
+
+Ngrok
+-----
+
+Ngrok is used in the development process to act as a tunnel into your PC. It is not secure to allow access into your PC
+or local address through a public channel, as this can open you up to malicious attacks. Ngrok allows you to securely
+provide access to your local address through something public, like the internet. When Ngrok is running on your machine,
+it gets the port of the network service, which is usually a web server, and then connects to the Ngrok cloud service.
+This cloud service takes in traffic and requests from a public address and then relays that traffic directly to the
+Ngrok process running on your machine, which then passes it along to the local address you specify. By doing this, you
+can minimize the interaction between outside traffic and your personal machine.
+
+When trying to stream videos in the development stage, the ember frameplayer will stream the video to PIPE's webservers,
+which will then write a request to AWS S3. A webhook will then send this video data back to the dummy url generated by ngrok,
+which then tunnels it to your local machine.
+
+
+
+.. When trying to stream videos on the development stage, AWS will SEND to PIPe need an address to send the video to. Ngrok will create
+a dummy link for this purpose and then send the video from this dummy address to your PC.
+
+
+.. From the deployed ember framplayer app the video is streamed to pipe's webservers, which forward/write request to video
+data to AWS S3 instance. Webhook (allows us to say send a payload to a certain address when u finish doing this) sends request
+back to url generated by ngrok which serves as a local tunnel to your computer since u cant point a webhook to local host
+
+.. In short, video data goes through this process: [Frameplayer app]--pipe client-->[Pipe service]--AWS storage trigger-->[AWS S3],
+then when the trigger finishes storing, it trips the “Finished uploading to S3” webhook that then sends a payload back
+to the lookit-API server, which has a handler that renames the file that was just stored in S3, among other things
+so that handler receives a payload from Pipe’s “finished uploading to S3 webhook”
+i.e., it’s Pipe telling lookit-API “hey, I finished putting some video in S3. Here’s some identifying data about that video.”
+
+
+
+
+
+
+
+External Resources
+___________________
+
+Google Cloud
+-------------
+
+The Cloud service is where all the code for the studies and
```

Comment:
unfinished sentence, and the start seems like something that would be more appropriate for the "how lookit works" section.

"Google Cloud" is also ambiguous; if you're talking about Google Cloud Platform that's also really broad - GCP has many services under its umbrella, including things we **do** use like Google Compute Engine, Google Cloud SQL, Google Kubernetes Engine,  and Google Cloud Storage (this last one is the one I think you're referring to), along with things that we **don't** use like Spanner and Bigtable.